### PR TITLE
Review request: Testing math-magician

### PR DESCRIPTION
**Pull request by chrystalme**
- Test all components
- Test calculate and operate logic

### CODE REVIEW

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- Good job by adding test units for functions in thecalculate.js :+1: 

To improve :factory_worker: [REQUIRED]:

- [x] [REQUIRED] all functionalities should be tested by using [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/), `@testing-library/react`, ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/testing/useReactTestingLibrary.png) from [Pathwright](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106580952/)).
- [x] [REQUIRED] Please add snapshot tests for Home, Calculator, and Quote.
- [x] [REQUIRED] Dont forget about negative tests.
- [x] [REQUIRED] All functions in calculate.js file are covered with unit tests (You are missing to test some of your calculator functionalities like error handling for example by dividing a number by 0).


That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Comment by chrystalme** [comment]
- Add snapshots test for Home, App and Quotes
- Use @testing-library/react for component testing
- Added negative test
- Catch error in calculate logic

File: src/__tests__/app-test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  });
+  test('renders button -', () => {
+    render(
+      <Routes>
+        <App />
+      </Routes>,
+    );
+    const buttonElement = screen.getByText(/-/i);
+    // eslint-disable-next-line no-unused-expressions
+    expect(buttonElement).toBeInTheDocument;
```
Comment (Elbie-em) [inline src/__tests__/app-test.js]: - [x] Kindly enable Eslint above and find a workaround fixing the lint error, it is meant as a guide to help you write clean and professional code following the programming standards.

File: src/__tests__/home-test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    const text = getByText('Math-magician');
+    // eslint-disable-next-line no-unused-expressions
+    expect(text).toBeInTheDocument;
+  });
+  test('Test Home page snapshot', () => {
+    const { getByText } = render(<Routes><Home /></Routes>);
+    const text = getByText('Math-magician');
+    expect(text).toMatchSnapshot();
+  });
+});
```
Comment (Elbie-em) [inline src/__tests__/home-test.js:18]: - [ ] Kindly enable Eslint above and find a workaround fixing the lint error, it is meant as a guide to help you write clean and professional code following the programming standards.

File: src/__tests__/quotes-test.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    expect(component).toMatchSnapshot();
+  });
+
+  test('Text match Home page test', () => {
+    const { getByText } = render(<Routes><Quotes /></Routes>);
+    const text = getByText('Math-magician');
+    // eslint-disable-next-line no-unused-expressions
+    expect(text).toBeInTheDocument;
+  });
+});
```
Comment (Elbie-em) [inline src/__tests__/quotes-test.js:21]: - [ ] Kindly enable Eslint above and find a workaround fixing the lint error, it is meant as a guide to help you write clean and professional code following the programming standards.

**Review by Elbie-em — APPROVED** [review]
## STATUS: APPROVED ✅ 

Hi @chrystalme, you have done a great job implementing the changes requested by the previous TSE and I am therefore approving this project. However, consider addressing the changes in my comments below ⬇️  before moving forward.

*All the best and keep up the good work* 💪

File: src/__tests__/quotes-test.js (hunk 3 above)
Comment (chrystalme) [reply src/__tests__/quotes-test.js:21]: Thanks for the suggestion. I had to import jest-dom from @testing

**Pull request merged**